src/app/data-explorer/querybuilder/querybuilder.component.html

```diff
@@ -123,6 +123,9 @@
             <mat-option
               *ngFor="let field of getFields(rule.entity)"
               [value]="field.value"
+              [matTooltip]="elementDescription(rule.entity, field.value)"
```

Comment:
I see a potential performance issue here because the template is calling a function to do input binding - but this is happening anyway because the `angular-querybuilder` library component is doing the same. The issue is that Angular change detection can no longer tell if this input binding has changed or not, so it is always going to be called and re-rendered. Confirm this by putting a `console.log()` line inside `elementDescription()` and viewing the results in the browser console.

The only way I can think of getting around this is to pre-cache all the descriptions. Here is some code I've tested which might work:

**querybuilder.component.ts**
```ts
descriptions: { [field: string]: string } = {};

ngOnInit() {
  this.dataElements.forEach((element) => {
      const entity = this.queryBuilderService.getEntity(element);
      const fullName = `${entity}.${element.label}`;

      const description = element.description
        ? element.description.length > 300
          ? element.description.substring(0, 300)
          : element.description
        : 'No additional information available';

      this.descriptions[fullName] = description;
    });
}
```

**querybuilder.component.html**
```html
<!-- Field Selector -->
      <ng-container
        *queryField="
          let rule;
          let fields = fields;
          let onChange = onChange;
          let getFields = getFields
        "
      >
        <mat-form-field
          appearance="outline"
          class="query-builder-field mat-form-field-no-padding"
        >
          <mat-select [(ngModel)]="rule.field" (ngModelChange)="onChange($event, rule)">
            <mat-option
              *ngFor="let field of getFields(rule.entity)"
              [value]="field.value"
              [matTooltip]="descriptions[rule.entity + '.' + field.value]"
              matTooltipClass="mdm-selection-item-tooltip"
              matTooltipPosition="right"
            >
              {{ field.name }}
            </mat-option>
          </mat-select>
        </mat-form-field>
      </ng-container>
```

Notice something similar is happening with `termSearchResults`. I'm not entirely sure if fetching from an array has the same affect as calling a function or not, but in general I think the change detection issue is avoidable if all data for binding is known up front.

Reply:
Updated as suggested. Yes, there was an issue with the old code constantly firing.